src/core/channel/http_client_filter.c

```diff
@@ -43,8 +43,9 @@ typedef struct call_data {
 
   int got_initial_metadata;
   grpc_stream_op_buffer *recv_ops;
-  void (*on_done_recv)(void *user_data, int success);
-  void *recv_user_data;
+  grpc_iomgr_closure *on_done_recv;
```

Comment:
The closure concept definitely improves readability and maintainability. Thanks.